src/asserts/null-or-uri-assert.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+const { Assert: BaseAssert, Violation } = require('validator.js');
+const UriAssert = require('./uri-assert');
+
+const Assert = BaseAssert.extend({
+  Uri: UriAssert
+});
+
+/**
+ * Export `NullOrUriAssert`.
+ */
+
+module.exports = function nullOrUriAssert() {
+  /**
+   * Class name.
+   */
+
+  this.__class__ = 'NullOrUri';
+
+  /**
+   * Validation algorithm.
+   */
+
+  this.validate = value => {
+    if (value !== null) {
+      try {
+        Assert.uri().validate(value);
+      } catch {
+        throw new Violation(this, value, { value: 'must_be_null_or_a_uri' });
```

Comment:
It would be best to use the original violation:

```suggestion
      } catch  (e) {
        throw new Violation(this, value, e.violation);
```